create_mattes_from_colored_mattes.py

```diff
@@ -0,0 +1,278 @@
+        sys.exit(1)
+
+
+def get_denoise_path(colormatte_path):
+    """Get the path for denoise frames. Input path must contain 'colormatte'."""
+    if 'colormatte' in colormatte_path:
+        denoise_path = colormatte_path.replace("colormatte", "denoise")
+
+        # Always use v001 for the denoise path. The number of frames will be the same in all versions.
+        denoise_index = denoise_path.find('denoise\\')
+        prefix = denoise_path[:denoise_index + len('denoise\\')]
+        
+        # Construct the new path with 'v001'
+        denoise_path = os.path.join(prefix, 'v001')
+        return denoise_path
+    else:
+        print("The provided colormattes must have a 'colormatte' folder in their path.")
+        sys.exit(1)
+
+
+def get_matte_filename(image_path):
+    """Get the matte filename from the input path"""
+    filename = os.path.basename(image_path)
+
+    prefix, frame, suffix = filename.split('.')
+    filename_without_extension = '.'.join([prefix, str(frame).zfill(8)]).replace("colormatte", "matte")
+
+    return filename_without_extension
+
+
+def get_matte_colors(image):
+    """Get the list of colors that were used for mattes in the given image."""
+    flattened_image = image.reshape((-1, 3))
+
+    unique_colors = np.unique(flattened_image, axis=0)
+    unique_colors_list = [tuple(color) for color in unique_colors]
+    black = (0, 0, 0)
+    unique_colors_list.remove(black)
+    return unique_colors_list
+
+
+def get_element_bw_matte_from_colormatte(image, target_color):
+    """Create a black and white matte from the area matted by the target color."""
+    # Convert the target color to a NumPy array
+    target_color_np = np.array(target_color, dtype=np.uint8)
+
+    # Create a b/w matte for the exact color using an equality check
+    matte = np.all(image == target_color_np, axis=-1).astype(np.uint8) * 255
+
+    return matte
+
+
+def create_black_matte(width, height):
+    """Create a solid black matte. Used for missing frames or missing element in a frame."""
+    return np.zeros((height, width, 3), dtype=np.uint8)
+
+
+def save_bw_matte(matte, element_matte_path, matte_filename):
+    """Save the black and white matte"""
+    os.makedirs(element_matte_path, exist_ok=True)
+
+    try:
+        cv2.imwrite(rf'{element_matte_path}\{matte_filename}.png', matte)
+        print(rf'Saved image {element_matte_path}\{matte_filename}.png')
+    except Exception as e:
+        print(f"Could not save file: {e}")
+
+
+def create_frame_dict(files):
+    """Create a lookup by frame number for a set of files."""
+    frame_dict = {}
+    for file in files:
+        frame_number = get_frame_number(file)
+        frame_dict[int(frame_number)] = file
+    return frame_dict
+
+
+parser = argparse.ArgumentParser(description='Create black and white mattes for elements in color mattes.')
+
+# Required parameters
+parser.add_argument('map_colors_path', type=str, help='Path to csv file that maps matte colors to elements')
```

Comment:
Perhaps since we are committing this to a public repo we should describe in a comment above or somewhere in here how this CSV is structured - ie, the columns and their expected values?